imap_processing/ultra/l1a/ultra_l1a.py

```diff
@@ -38,13 +41,70 @@ def xarray_aux(decom_ultra_aux: dict):
         attrs=ConstantCoordinates.EPOCH,
     )
 
-    dataset = xr.Dataset(
-        coords={"Epoch": epoch_time},
-        attrs=ultra_cdf_attrs.ultra_l1a_attrs.output(),
-    )
+    if apid != ULTRA_TOF.apid[0]:
+        dataset = xr.Dataset(
+            coords={"Epoch": epoch_time},
+            attrs=ultra_cdf_attrs.ultra_l1a_attrs.output(),
+        )
+    else:
+        row = xr.DataArray(
+            # Number of pixel rows
+            np.arange(54),
+            name="Row",
+            dims=["Row"],
+            attrs=dataclasses.replace(
+                ultra_cdf_attrs.ultra_support_attrs,
+                catdesc="ROW",  # TODO: short and long descriptions
+                fieldname="ROW",
+                var_type="ignore_data",
+            ).output(),
+        )
 
-    for key, value in decom_ultra_aux.items():
-        if key in [
+        column = xr.DataArray(
+            # Number of pixel columns
+            np.arange(180),
+            name="Column",
+            dims=["Column"],
+            attrs=dataclasses.replace(
+                ultra_cdf_attrs.ultra_support_attrs,
+                catdesc="COLUMN",  # TODO: short and long descriptions
+                fieldname="COLUMN",
+                var_type="ignore_data",
+            ).output(),
+        )
+
+        dataset = xr.Dataset(
+            coords={"Epoch": epoch_time, "Row": row, "Column": column},
+            attrs=ultra_cdf_attrs.ultra_l1a_attrs.output(),
+        )
+
+    return dataset
+
+
+def xarray(decom_ultra: dict, apid: int):
```

Comment:
This would shadow `xarray` itself so causes some confusion. Adding a verb to the function would help here.
```suggestion
def create_xarray(decom_ultra: dict, apid: int):
```